pkg/store/prometheus.go

```diff
@@ -330,7 +336,7 @@ func (p *PrometheusStore) handleSampledPrometheusResponse(
 	span.SetTag("series_count", len(resp.Results[0].Timeseries))
 
 	for _, e := range resp.Results[0].Timeseries {
-		lset := labelpb.ExtendSortedLabels(labelpb.ZLabelsToPromLabels(e.Labels), extLset)
+		lset := labelpb.ExtendSortedLabels(labelpb.ZLabelsToPromLabels(e.Labels), rmLabels(extLset.Copy(), extLsetToRemove))
```

Comment:
Copy is needed because replica labels could be set per-request :cry:

Reply:
I think this is fine, it's only copying the slice without making a deep copy.